drs_downloader/cli.py

```diff
@@ -95,44 +102,102 @@ def _extract_tsv_info(manifest_path_: Path, drs_header: str = 'pfb:ga4gh_drs_uri
         Returns:
             List[str]: The URI's corresponding to the DRS objects.
         """
-        uris = []
+        urls = []
+        uri_index = 0
+        header = None
+        with open(Path(manifest_path_)) as file:
+            tsv_file = csv.reader(file, delimiter="\t")
+            headers = next(tsv_file)
+
+            # search for header name
+            if drs_header is None:
+                for col in headers:
+                    if ('uri' in col.lower()):
+                        header = headers[headers.index(col)]
+                        uri_index = headers.index(col)
+                        break
+            else:
+                if(drs_header in headers):
+                    uri_index = headers.index(drs_header)
+                    header = drs_header
+
+            # add url to urls list
+            if (header is not None):
+                for row in tsv_file:
+                    urls.append(row[uri_index])
+            else:
+                raise KeyError("Key format for drs_uri is bad. Make sure the collumn that contains the URIS \
+                has 'uri' somewhere in it, or the URI header matches the uri header name in the TSV file that was specified")
+            
+            for url in urls:
+                if('/' in url):continue
+                else:
+                    raise Exception("Check that your header name for your DRS URIS is directly above the column of your DRS URIS")
+            return urls
 
-        df = pd.read_csv(manifest_path_, sep='\t')
-        if drs_header in df.columns.values.tolist():
-            for i in range(df[drs_header].count()):
-                uris.append(df['pfb:ga4gh_drs_uri'][i])
-        else:
-            raise KeyError(f"Header '{drs_header}' not found in {manifest_path}")
 
-        return uris
+    print("THE VALUE OF MANIFEST PATH",manifest_path)
+    ids_from_manifest = _extract_tsv_info(manifest_path,drs_header)
 
-    ids_from_manifest = _extract_tsv_info(manifest_path)
 
-    # get the DrsObject for the ids
     drs_objects = drs_manager.get_objects(ids_from_manifest)
+    drs_objects.sort(key=lambda x: x.size, reverse=False)
+
+    print("THE VALUE OF DRS OBJECTS ",drs_objects[0])    
+    total_batches = len(ids_from_manifest) / DEFAULT_MAX_SIMULTANEOUS_OBJECT_SIGNERS
+
+    if (math.ceil(total_batches) - total_batches >  0):
+        total_batches += 1
+        total_batches = int(total_batches)
+
+    ids_list = [objects.id for objects in drs_objects]
+    #size_list = [objects.size for objects in drs_objects]
+
+    #for object.id in drs_objects
+    start =0
+    for chunk_of_object_ids in tqdm.tqdm(
+                DrsAsyncManager._chunker(ids_list, DEFAULT_MAX_SIMULTANEOUS_OBJECT_SIGNERS),
```

Comment:
why is this now a list of ids instead of a list of populated drs objects (which include the id)?